pandas/core/arrays/datetimelike.py

```diff
@@ -359,7 +359,14 @@ def __array__(
     ) -> np.ndarray:
         # used for Timedelta/DatetimeArray, overwritten by PeriodArray
         if is_object_dtype(dtype):
+            if copy is False:
+                raise ValueError(
+                    "Unable to avoid copy while creating an array as requested."
+                )
             return np.array(list(self), dtype=object)
+
+        if copy is True:
+            return np.array(self._ndarray, dtype=dtype)
         return self._ndarray
```

Comment:
Does it matter much for the numpy side whether we return the ndarray here directly (which might not match the dtype), or already do `return np.asarray(self._ndarray, dtype=dtype)` on our side?

Reply:
No, NumPy will deal with dtype mismatch.  It's been historically more of a hint and not a requirement.